### ISSUE TITLE
Unable to create web-worker if SharedArrayBuffer is enabled

### ISSUE THREAD

**Issue by LeXXik** [playcanvas/editor#1103]
Unable to create a web-worker, if SharedArrayBuffer option is enabled.

Repro: https://playcanvas.com/project/1191623/overview/corp-issue

![image](https://github.com/playcanvas/engine/assets/5677782/2f27fd94-0e1b-4302-a7a1-f2ff213771c6)

![image](https://github.com/playcanvas/engine/assets/5677782/06830e4e-7001-4305-976f-c44afdb25fe9)

**Comment by MAG-AdrianMeredith** [comment]
Literally came here to log the same issue.  We have a client hoping to make use of this.  It looks like assets come from a different server and that one doesn;t set the COOP + COEP headers that are required for secure features e.g. sharedarraybuffer and workers

**Comment by mvaligursky** [comment]
Do we know what the solution to this is?
@isumygin-sc or anybody else?

**Comment by MAG-AdrianMeredith** [comment]
@mvaligursky It should be the same that you did to add SharedArrayBuffer support.  All of the servers you use need to return COOP and COEP http headers.
Basically do exactly what that error message says in the screenshot above

**Comment by isumygin-sc** [comment]
We use **credentialless** for Cross-Origin-Embedder-Policy if SharedArrayBuffer is enabled

[require-corp](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Embedder-Policy#require-corp)
A document can only load resources from the same origin, or resources explicitly marked as loadable from another origin. If a cross origin resource supports CORS, the [crossorigin](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/crossorigin) attribute or the [Cross-Origin-Resource-Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Resource-Policy) header must be used to load it without being blocked by COEP.

[credentialless](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Embedder-Policy#credentialless)
[no-cors](https://developer.mozilla.org/en-US/docs/Web/API/Request/mode) cross-origin requests are sent without credentials. In particular, it means Cookies are omitted from the request, and ignored from the response. The responses are allowed without an explicit permission via the [Cross-Origin-Resource-Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Resource-Policy) header. [Navigate](https://developer.mozilla.org/en-US/docs/Web/API/Request/mode) responses behave similarly as the require-corp mode: They require [Cross-Origin-Resource-Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Resource-Policy) response header.

**Comment by isumygin-sc** [comment]
so the headers are set for the launch page only - how do you load a worker?

**Comment by isumygin-sc** [comment]
could you build me a test case please?

**Comment by marklundin** [comment]
https://playcanvas.com/editor/scene/1953023

**Comment by yak32** [comment]
i prepared a fix, will be deployed soon

**Comment by LeXXik** [comment]
Eagerly waiting ;)

**Comment by willeastcott** [comment]
@LeXXik It should be Monday morning. 😄

**Comment by isumygin-sc** [comment]
the fix is deployed, please test your projects

**Comment by LeXXik** [comment]
It works on my end. Thank you!

Although, it throws an issue with Draco. I will open another ticket for it.
